sdk/keyvault/azsecrets/custom_client.go

```diff
@@ -0,0 +1,39 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azsecrets
+
+// this file contains handwritten additions to the generated code
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/internal"
+)
+
+// NewClient creates a client that accesses a Key Vault's secrets.
+func NewClient(vaultURL string, credential azcore.TokenCredential, options *azcore.ClientOptions) *Client {
+	authPolicy := internal.NewKeyVaultChallengePolicy(credential)
+	pl := runtime.NewPipeline(moduleName, version, runtime.PipelineOptions{PerRetry: []policy.Policy{authPolicy}}, options)
+	return &Client{endpoint: vaultURL, pl: pl}
+}
+
+// ID is a secret's unique ID, containing its name and version.
+type ID string
+
+// Name of the secret.
+func (i ID) Name() string {
```

Comment:
Since `ID` is returned by pointer, this will dereference `i` and could panic if it's ever `nil`.  It might be worth changing the receivers to byref and just return the empty string if `i` is `nil`.

Reply:
I don't feel too strong about this though, and you could make the argument that `ID` is never `nil` so if you do get a panic then you likely have a bug in your code (I'm somewhat guessing on this latter part, let me know if you disagree).

Reply:
I'm OK with panics for developer bugs. But since it is returned by pointer, these probably should be pointers too.

Reply:
Thanks for pointing this out, I'll make them pointer receivers. I want to keep the value return because every secret has an ID and Key Vault always includes it in responses so, without a bug we would want to know about or some other extraordinary circumstance, `ID` won't be `nil`.